web-ui/src/streaming/introspection/IntrospectionTable.tsx

```diff
@@ -0,0 +1,128 @@
+// Browse the contents of a table or a view.
+
+import Card from '@mui/material/Card'
+import { DataGridPro, GridColumns, useGridApiRef } from '@mui/x-data-grid-pro'
+import { useQuery } from '@tanstack/react-query'
+import { useEffect, useState } from 'react'
+import { PipelineDescr, ProgramDescr, PipelineStatus, OpenAPI } from 'src/types/manager'
+import { parse } from 'csv-parse'
+import { parseProjectSchema } from 'src/types/program'
+
+export type IntrospectionTableProps = {
+  pipelineDescr: PipelineDescr | undefined
+  name: string | undefined
+}
+
+export const IntrospectionTable = ({ pipelineDescr, name }: IntrospectionTableProps) => {
+  const apiRef = useGridApiRef()
+  const [headers, setHeaders] = useState<GridColumns | undefined>(undefined)
+
+  // Lookup the schema of the program
+  const projectQuery = useQuery<ProgramDescr>(['programStatus', { program_id: pipelineDescr?.program_id }], {
+    enabled: pipelineDescr !== undefined && pipelineDescr.program_id !== undefined
+  })
+  useEffect(() => {
+    if (!projectQuery.isLoading && !projectQuery.isError && name) {
+      if (projectQuery.data && projectQuery.data.schema) {
+        const program = parseProjectSchema(projectQuery.data)
+        const view = program.schema['outputs'].find(v => v.name === name)
+
+        if (view) {
+          const id = [{ field: 'genId', headerName: 'genId' }]
```

Comment:
what's this for?

Reply:
it's an identifier for the row so I know what row to update when a change comes in